lms/djangoapps/discussion_api/tests/test_views.py

```diff
@@ -201,3 +201,120 @@ def test_pagination(self):
+        super(CommentViewSetListTest, self).setUp()
+        self.author = UserFactory.create()
+        self.url = reverse("comment-list")
+        self.thread_id = "test_thread"
+
+    def test_thread_id_missing(self):
+        response = self.client.get(self.url)
+        self.assert_response_correct(
+            response,
+            400,
+            {"field_errors": {"thread_id": "This field is required."}}
+        )
+
+    def test_404(self):
+        self.register_get_thread_error_response(self.thread_id, 404)
+        response = self.client.get(self.url, {"thread_id": self.thread_id})
+        self.assert_response_correct(
+            response,
+            404,
+            {"developer_message": "Not found."}
+        )
+
+    def test_basic(self):
+        self.register_get_user_response(self.user, upvoted_ids=["test_comment"])
+        source_comments = [{
+            "id": "test_comment",
+            "thread_id": self.thread_id,
+            "parent_id": None,
+            "user_id": str(self.author.id),
+            "username": self.author.username,
+            "anonymous": False,
+            "anonymous_to_peers": False,
+            "created_at": "2015-05-11T00:00:00Z",
+            "updated_at": "2015-05-11T11:11:11Z",
+            "body": "Test body",
+            "abuse_flaggers": [],
+            "votes": {"up_count": 4},
+            "children": [],
+        }]
+        expected_comments = [{
+            "id": "test_comment",
+            "thread_id": self.thread_id,
+            "parent_id": None,
+            "author": self.author.username,
+            "author_label": None,
+            "created_at": "2015-05-11T00:00:00Z",
+            "updated_at": "2015-05-11T11:11:11Z",
+            "raw_body": "Test body",
+            "abuse_flagged": False,
+            "voted": True,
+            "vote_count": 4,
+            "children": [],
+        }]
+        self.register_get_thread_response({
+            "id": self.thread_id,
+            "course_id": unicode(self.course.id),
+            "thread_type": "discussion",
+            "children": source_comments,
+            "resp_total": 100,
+        })
+        response = self.client.get(self.url, {"thread_id": self.thread_id})
+        self.assert_response_correct(
+            response,
+            200,
+            {
+                "results": expected_comments,
+                "next": "http://testserver/api/discussion/v1/comments/?thread_id={}&page=2".format(
+                    self.thread_id
+                ),
+                "previous": None,
+            }
+        )
+        self.assert_query_params_equal(
+            httpretty.httpretty.latest_requests[-2],
+            {
+                "recursive": ["True"],
+                "resp_skip": ["0"],
+                "resp_limit": ["10"],
+                "user_id": [str(self.user.id)],
+                "mark_as_read": ["True"],
+            }
+        )
+
+    def test_pagination(self):
```

Comment:
Looks like this is testing pagination for when trying to access a page greater than the number of pages. Since the name "test_pagination" isn't clear, adding a doc string would help.